lib/elixir/lib/enum.ex

```diff
@@ -3856,8 +3861,13 @@ defimpl Enumerable, for: Map do
   end
 
   def slice(map) do
-    size = map_size(map)
-    {:ok, size, &Enumerable.List.slice(:maps.to_list(map), &1, &2, size)}
+    case map_size(map) do
+      0 ->
+        {:ok, 0, fn _, _ -> [] end}
```

Comment:
the documentation for `t:Enumerable.slicing_fun/0` needs to be updated, since the slicing function can return an empty list, as shown here, and the element is not guaranteed to be in the enumerable if the enumerable is empty.

> slicing_fun() ::
>   (start :: non_neg_integer(), length :: pos_integer() -> [term()])
> 
> A slicing function that receives the initial position and the number of elements in the slice.
> 
> The start position is a number >= 0 and guaranteed to exist in the enumerable. The length is a number >= 1 in a way that start + length <= count, where count is the maximum amount of elements in the enumerable.
> 
> The function should return a non empty list where the amount of elements is equal to length.
>